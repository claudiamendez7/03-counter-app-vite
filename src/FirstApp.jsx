// export function App() {
//     return (<h1>Hola Mundo</h1>);
// }

// export default App;

// import React from 'react'

export const FirstApp = () => {

    
    return (
        <h1>FirstApp</h1>
    );
}

// export const FirstApp = () =><h1>Hello World</h1>;
