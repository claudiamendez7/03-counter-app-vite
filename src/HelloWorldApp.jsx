// import {fragment} from 'react'; antigua importacion

// const getResult = () => {
//     return 5*8;
// }

// const newMessage = {
//     message: 'Batman',
//     title: 'Claudia'
// };

import PropTypes from 'prop-types';

export const HelloWorldApp = ({title, subTitle, name}) => {    
    // console.log(props);
    // if ({title}) {
    //     throw new Error('El tile no existe');
    // }
    return (
        <>
            <h1>{title}</h1>
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    )
}

HelloWorldApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

HelloWorldApp.defaultProps = {
    title :'No hay titulo',
    subTitle : 'No hay subtitulo',
    name: 'Claudia Mendez'
}
