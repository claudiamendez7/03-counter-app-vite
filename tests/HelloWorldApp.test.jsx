import { render } from "@testing-library/react";
import { HelloWorldApp } from "../src/HelloWorldApp";


describe('Pruebas en <HelloWorldApp/>', () => {

    test('Debe de hacer match con el snapshot', () => {

        const title = 'Hola, Soy Claus';
        const {container} = render(<HelloWorldApp title={title}/>);

        expect (container).toMatchSnapshot();

    });

    test('Debe mostrar el titulo de un h1', () => {

        const title = 'Hola, Soy Claus';
        const {container, getByText} = render(<HelloWorldApp title={title}/>);

        expect (getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);

    });

});