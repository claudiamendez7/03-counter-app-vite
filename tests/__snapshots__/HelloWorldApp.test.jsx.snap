// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HelloWorldApp/> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Claus
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Claudia Mendez
  </p>
</div>
`;
